main.tf

```diff
@@ -0,0 +1,208 @@
+locals {
+  tags = "${merge(var.tags, map("kubernetes.io/cluster/${var.cluster_name}", "owned"))}"
+
+  # userdata for EKS worker nodes to properly configure Kubernetes applications on EC2 instances
+  # https://docs.aws.amazon.com/eks/latest/userguide/launch-workers.html
+  userdata = <<USERDATA
+  #!/bin/bash
+  set -o xtrace
+  /etc/eks/bootstrap.sh --apiserver-endpoint '${var.cluster_endpoint}' --b64-cluster-ca '${var.cluster_certificate_authority_data}' '${var.cluster_name}'
```

Comment:
This script assumes some specific AMIs?

Reply:
yes, EKS AMI
https://docs.aws.amazon.com/eks/latest/userguide/eks-optimized-ami.html